Detectors/ITSMFT/common/workflow/src/STFDecoderSpec.cxx

```diff
@@ -112,22 +112,32 @@ void STFDecoder<Mapping>::run(ProcessingContext& pc)
   std::vector<o2::itsmft::CompClusterExt> clusCompVec;
   std::vector<o2::itsmft::ROFRecord> clusROFVec;
   std::vector<unsigned char> clusPattVec;
-  std::vector<Digit> digVec;
+  std::vector<o2::itsmft::DigitHW> digVecHW;
+  std::vector<o2::itsmft::Digit> digVec;
   std::vector<ROFRecord> digROFVec;
   mDecoder->setDecodeNextAuto(false);
   while (mDecoder->decodeNextTrigger()) {
     if (mDoDigits) {                                    // call before clusterization, since the latter will hide the digits
-      mDecoder->fillDecodedDigits(digVec, digROFVec);   // lot of copying involved
+      if (mWriteHW) {
+        mDecoder->fillDecodedDigitsHW(digVecHW, digROFVec); // lot of copying involved
+      } else {
+        mDecoder->fillDecodedDigits(digVec, digROFVec); // lot of copying involved
+      }
     }
     if (mDoClusters) { // !!! THREADS !!!
       mClusterer->process(mNThreads, *mDecoder.get(), &clusCompVec, mDoPatterns ? &clusPattVec : nullptr, &clusROFVec);
     }
   }
 
   if (mDoDigits) {
-    pc.outputs().snapshot(Output{orig, "DIGITS", 0, Lifetime::Timeframe}, digVec);
+    if (mWriteHW) {
+      pc.outputs().snapshot(Output{orig, "DIGITS", 0, Lifetime::Timeframe}, digVecHW);
+    } else {
+      pc.outputs().snapshot(Output{orig, "DIGITS", 0, Lifetime::Timeframe}, digVec);
+    }
```

Comment:
It is error-prone sending different  types under the same DataDescription

Reply:
Does it mean that you are not in favour of any conditional switch in the code?